src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -418,16 +418,17 @@ public async Task Ctor_SafeHandle_Listening_Success(bool shareSafeHandle)
             using var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
             listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
             listener.Listen();
-            Assert.Equal(1, listener.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.AcceptConnection));
```

Comment:
Should we fix this to work correctly on mac, too?

Reply:
I don't understand the comment @stephentoub. I don't think this particular options is supported by `getsockopt()` on OSX.

Reply:
> I don't think this particular options is supported by getsockopt() on OSX.

Exactly; I'm asking if we should make it work.  We already do work in the PAL to smooth over some options:
https://github.com/dotnet/runtime/blob/1cfccd2a7937117341b256847e613dd710a70adb/src/libraries/Native/Unix/System.Native/pal_networking.c#L1908-L1911
I'm asking whether you think we should add use of `proc_pidfdinfo` there to make `listener.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.AcceptConnection)` work on macOS as it does on Windows and Linux.

Reply:
yes, I think so.

Reply:
updated + one new test.